crates/plugins/mediator-coordination/src/repository/stateful.rs

```diff
@@ -317,29 +112,14 @@ pub mod tests {
                             return false;
                         }
                     }
-
-                    if let Some(client_did) = filter.get("client_did") {
-                        if json!(c.client_did) != json!(client_did) {
-                            return false;
-                        }
-                    }
-
```

Comment:
You removed this section in the new MockConnectionRepository implementation. Tests fail because the match on `client_did` becomes lazy.